lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -126,9 +113,9 @@ protected function getFilename($id)
     {
         return $this->directory
             . DIRECTORY_SEPARATOR
-            . implode(str_split(hash('sha256', $id), 2), DIRECTORY_SEPARATOR)
+            . substr(hash('sha256', $id), 0, 2)
```

Comment:
Two hex chars to balance the tree are enough and is the best compromise. This is also used by twig now.
Files will be evenly distributed between 16^2 = 256 directories.

The old solution created unique directory tree for each file. For each file it created 32 directories! Totally unusable.